crates/wasmtime/src/trap.rs

```diff
@@ -226,4 +221,37 @@ impl fmt::Display for Trap {
     }
 }
 
-impl std::error::Error for Trap {}
+impl std::error::Error for Trap {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match &self.inner.reason {
+            TrapReason::Error(e) => Some(e.as_ref()),
+            TrapReason::I32Exit(_) | TrapReason::Message(_) => None,
+        }
+    }
+}
+
+impl From<anyhow::Error> for Trap {
+    fn from(e: anyhow::Error) -> Trap {
+        match e.downcast::<Trap>() {
```

Comment:
I don't think this quite what we want because the semantics of `anyhow` is that it searches the entire chain of errors and returns `Trap` if anything is a `Trap`, which means that this can lose context.

Reply:
Good point, I was not aware of this. I made this impl delegate to the `Box<dyn Error + Send + Sync>` one, and used the std error `downcast_ref` there.